### DISCUSSION TITLE
ArchR::mergeGR() for blacklist files

### DISCUSSION THREAD

**Discussion by wuv21** [GreenleafLab/ArchR#723] [Questions / Documentation]
Hi! Thank you for an excellent package and for making scATACseq so much more accessible (pun intended).

I noticed in the documentation (see below for excerpt) that the blacklist file is made by merging the two bed/narrowPeak files with `ArchR::mergeGR()`. However, this function doesn't seem to exist in the package (or in the hidden/private functions) that I can see. Would it be possible to detail how those two bed/narrowPeak files were merged or for a link to the function?

> The precompiled version of the hg38 genome in ArchR uses BSgenome.Hsapiens.UCSC.hg38, TxDb.Hsapiens.UCSC.hg38.knownGene, org.Hs.eg.db, and a blacklist that was merged using ArchR::mergeGR() from the hg38 v2 blacklist regions and from mitochondrial regions that show high mappability to the hg38 nuclear genome from Caleb Lareau and Jason Buenrostro. To set a global genome default to the precompiled hg38 genome:

Thanks in advance!

**Comment by rcorces** [comment]
Thats a good question. To be honest, I dont know the answer but my intuition is that a `mergeGR` function once existed and was removed at some point during beta. My best guess is that the following function is approximately what was used. @jgranja24 - any insight?

```
mergeGRanges <- function(gr, ignore.strand = TRUE){
  gr <- validGRanges(gr)
  grR <- reduce(gr,min.gapwidth=0L,ignore.strand = ignore.strand)
  o <- DataFrame(findOverlaps(grR, gr,ignore.strand = ignore.strand))
  o$start <- start(gr[o$subjectHits])
  o$end <- end(gr[o$subjectHits])
  o$chr <- seqnames(gr[o$subjectHits])
  os <- o[order(o$start,decreasing = FALSE),] %>% {.[!duplicated(.$queryHits),c("queryHits", "start")]}
  oe <- o[order(o$end,decreasing = TRUE),] %>% {.[!duplicated(.$queryHits),c("queryHits", "end")]}
  oc <- o[!duplicated(o$queryHits),c("queryHits", "chr")]
  df <- merge(merge(oc, os, by = "queryHits"),oe,  by = "queryHits")
  mGR <- GRanges(df[,2], ranges = IRanges(df[,3], df[,4])) %>% sortSeqlevels %>% sort
  return(mGR)
}
```
